### DISCUSSION TITLE
Importing a custom robot

### DISCUSSION THREAD

**Discussion by ge79puv** [StanfordVL/BEHAVIOR-1K#993] [Q&A] [answered]
Hey there,

I am trying to import kuka iiwa 7 into the OmniGibson assets. I already complete the first steps about urdf in https://behavior.stanford.edu/omnigibson/tutorials/custom_robot_import.html. However, I notice that I also need to provide a yaml file for the manipulation robot. Should I use the Lula Robot Description Editor in Isaac Sim and the kuka urdf to create the robot_descriptor.yaml? Or maybe there is other method to do that? 

Best regards,

Yunfeng

**Comment by wensi-ai** [comment]
Hi @ge79puv,

The robot import tutorial is actually a bit outdated. We've updated the IK controller so there's no need to create the yaml descriptor file now.

**Reply by ge79puv** [reply]
Hi @wensi-ai ,

thanks for your rapid reply. However, when I run the demo of omnigibson.examples.robots.robot_control_example with my imported robot, there is an error about the missing yaml file. I use OmniGibson v1.1.1 and Isaac Sim 4.1.0. I also don't want to create the robot_descriptor.yaml, could you help me to fix this issue?

![image](https://github.com/user-attachments/assets/57f66b49-88f3-4a66-bc5c-fcb646ba9548)

**Reply by cgokmen** [reply]
Thanks for pointing this out, we will patch this issue asap.

**Comment by wensi-ai** [comment]
Hi @ge79puv , 

Seems like we didn't fully clean up our legacy code. For now you can just add a dummy implementation of the yaml file to your robot class like [this one](https://github.com/StanfordVL/OmniGibson/blob/main/omnigibson/robots/a1.py#L195) (notice that the file actually doesn't exist in the asset), and it should work for your setting. We will fix the bug in our next release. Thanks for pointing this out!

**Reply by ge79puv** [reply]
Hi @wensi-ai ,

you are right, I can just use a dummy yaml file, and that solves my problem. 

Btw, do I need to complete the functions of def assisted_grasp_start_points and def assisted_grasp_end_points? And there are some small mistakes in these two functions in vx300s, a1 and other robot .py files. assisted_grasp_end_points returns the same value as assisted_grasp_start_points. 

Best regards, 

Yunfeng

**Reply by wensi-ai** [reply, marked as answer]
Those are only required if you want to use assisted grasping for your project, so you can ignore them if you just want to use physical grasp or sticky grasp. I don't think vx300s and a1 returns the same value for these (notice the points are defined w.r.t. different eef links). For more information you can checkout the grasping demo we provide, or the `Optional properties` under `Create the Robot Class` in the create robot tutorial on how to defined these values.
